src/Domain/Creators/BinaryCompactObjectHelpers.hpp

```diff
@@ -63,6 +65,84 @@ std::unordered_map<std::string, tnsr::I<double, 3, Frame::Grid>>
 create_grid_anchors(const std::array<double, 3>& center_a,
                     const std::array<double, 3>& center_b);
 
+template <domain::ObjectLabel Object>
+struct ShapeMapOptions {
+  static std::string name() { return "ShapeMap" + get_output(Object); }
+  static constexpr Options::String help = {
+      "Options for a time-dependent distortion (shape) map about the "
+      "specified object. Specify 'None' to not use this map."};
+
+  struct LMax {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "LMax used for the number of spherical harmonic coefficients of the "
+        "distortion map. Currently, all coefficients are initialized to "
+        "zero."};
+  };
+
+  struct SizeInitialValues {
+    using type = std::array<double, 3>;
+    static constexpr Options::String help = {
+        "Initial value and two derivatives of the size map."};
+  };
+
+  using options = tmpl::list<LMax, SizeInitialValues>;
+
+  size_t l_max{};
+  std::array<double, 3> initial_size_values{};
+};
+
+namespace detail {
+// Convenience type alias
+template <typename... Maps>
+using gi_map = domain::CoordinateMap<Frame::Grid, Frame::Inertial, Maps...>;
+template <typename... Maps>
+using gd_map = domain::CoordinateMap<Frame::Grid, Frame::Distorted, Maps...>;
+template <typename... Maps>
+using di_map =
+    domain::CoordinateMap<Frame::Distorted, Frame::Inertial, Maps...>;
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+struct produce_all_maps;
+
+// Case to end recursion. No maps means empty list
+template <typename SourceFrame, typename TargetFrame>
+struct produce_all_maps<SourceFrame, TargetFrame, tmpl::list<>> {
+  using type = tmpl::list<>;
+};
+
+/*
+ * This will produce all the possible combinations of maps we will need for the
+ * maps_list. It does so through recursion. Say you have 3 maps, Map1, Map2,
+ * Map3. The end result of this will be a list with the following map
+ * combinations where these combinations may not appear in the same position in
+ * the list, but the order of each combination is fixed:
+ *  - Map1
+ *  - Map2
+ *  - Map3
+ *  - Map1, Map2
+ *  - Map1, Map3
+ *  - Map2, Map3
+ *  - Map1, Map2, Map3
+ */
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+struct produce_all_maps<SourceFrame, TargetFrame, tmpl::list<Maps...>> {
```

Comment:
I gave you a better algorithm for this elsewhere.  Use that instead.

Reply:
```c++
template <typename SourceFrame, typename TargetFrame, typename Maps>
using produce_all_maps_helper =
    tmpl::wrap<tmpl::push_front<Maps, SourceFrame, TargetFrame>,
               domain::CoordinateMap>;
template <typename SourceFrame, typename TargetFrame, typename... Maps>
using produce_all_maps_t = tmpl::transform<
    tmpl::remove<typename power_set<tmpl::list<Maps...>>::type, tmpl::list<>>,
    tmpl::bind<produce_all_maps_helper, tmpl::pin<SourceFrame>,
               tmpl::pin<TargetFrame>, tmpl::_1>>;
```
The parameter pack makes it a little tricky.

The "_t" name isn't appropriate anymore, of course.

Reply:
Ok yup this worked!! Pasting the `power_set` implementation you posted in slack here for completeness:

```cpp
template <typename List>
struct power_set {
  using rest = typename power_set<tmpl::pop_front<List>>::type;
  using type = tmpl::append<
      rest, tmpl::transform<rest, tmpl::lazy::push_front<
                                      tmpl::_1, tmpl::pin<tmpl::front<List>>>>>;
};

template <>
struct power_set<tmpl::list<>> {
  using type = tmpl::list<tmpl::list<>>;
};
```